Sources/ExyteChat/Model/Attachment.swift

```diff
@@ -29,23 +29,70 @@ public enum AttachmentType: String, Codable, Sendable {
 }
 
 public struct Attachment: Codable, Identifiable, Hashable, Sendable {
+    
+    public enum UploadStatus: Sendable, Codable, Hashable {
+        case inProgress(Int?) // value = percent upto 99%, nil no percent shown, progress indicator only
+        case complete
+        case cancelled
+        case error
+        
+        public static func == (lhs: UploadStatus, rhs: UploadStatus) -> Bool {
+            switch (lhs, rhs) {
+            case (.error, .error):
+                return true
+            case (.complete, .complete):
+                return true
+            case (.cancelled, .cancelled):
+                return true
+            case (.inProgress(let lhsPercent), .inProgress(let rhsPercent)):
+                return lhsPercent == rhsPercent
+            default:
+                return false
+            }
+        }
+    }
+    
+    
     public let id: String
     public let thumbnail: URL
     public let full: URL
+    public let fullUploadStatus: UploadStatus?
     public let type: AttachmentType
     public let thumbnailCacheKey: String?
     public let fullCacheKey: String?
 
-    public init(id: String, thumbnail: URL, full: URL, type: AttachmentType, thumbnailCacheKey: String? = nil, fullCacheKey: String? = nil) {
+    public init(id: String, thumbnail: URL, full: URL, type: AttachmentType, thumbnailCacheKey: String? = nil,
+                fullCacheKey: String? = nil, fullUploadStatus: UploadStatus? = nil) {
```

Comment:
UploadStatus defaults to nil, which results in no change for how attachments are handled.